contracts/modules/cover/Quotation.sol

```diff
@@ -69,44 +69,73 @@ contract Quotation is Iupgradable {
   }
 
   /**
-   * @dev Expires a cover after a set period of time.
-   * Changes the status of the Cover and reduces the current
-   * sum assured of all areas in which the quotation lies
-   * Unlocks the CN tokens of the cover. Updates the Total Sum Assured value.
-   * @param _cid Cover Id.
+   * @dev Expires a cover after a set period of time and changes the status of the cover
+   * @dev Reduces the total and contract sum assured
+   * @param coverId Cover Id.
    */
-  function expireCover(uint _cid) public {
-    require(checkCoverExpired(_cid) && qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.CoverExpired));
+  function expireCover(uint coverId) external {
 
-    tf.unlockCN(_cid);
-    bytes4 curr;
-    address scAddress;
-    uint sumAssured;
-    (,, scAddress, curr, sumAssured,) = qd.getCoverDetailsByCoverID1(_cid);
-    if (qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.ClaimAccepted))
-      _removeSAFromCSA(_cid, sumAssured);
-    qd.changeCoverStatusNo(_cid, uint8(QuotationData.CoverStatus.CoverExpired));
+    uint expirationDate = qd.getValidityOfCover(coverId);
+    require(expirationDate < now, "Quotation: cover is not due to expire");
```

Comment:
why is it ok to expire the cover before the grace period is passed?

Reply:
We expire the cover to release the capacity.